Prototyping Machine learning models with streamlit

jamessandy opened the pull request:
### Article Checklist
#### Preliminary Checks - Formatting and Structure 
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
- [x] In your PR - add a comment that states **["This fixes #(enter your Idea Content Form #)"](https://youtu.be/TKJ4RdhyB5Y?t=278)** to link and close your Idea Content (issue) when the PR is merged.
      

#### Grammar Checks
- [x] Have you spellchecked and grammar checked your article? - **Tool:** [Grammarly](https://grammarly.com)
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical Checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**

### Contribution Guidelines
For first-time contributors and for more details, **see [Contributing Guidelines](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md)**

jamessandy commented:
This fixes #499

jamessandy commented:
@lalith1403

hectorkambow commented:
Plagiarism check done.

hectorkambow commented:
@jamessandy @lalith1403  
This PR has merge conflicts. 
@jamessandy - lets be sure we are forking the repo correctly. 
PRs should not have merge issues upon submission.

hectorkambow commented:
Deleted author page - as it seemed to create a merge conflict. 
The avatar image can be deleted as well.

jamessandy commented:
Deleted

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,185 @@
+#### Building Sentiment Analyzer
+Now for this introductory tutorial, we will be building a sentiment analyzer model and prototype it with streamlit. The first step will be getting a Dataset, and the dataset we will be using can be found HERE, next let’s create a directory and create a virtual environment.
+
+```python
````
Specify where one should type in these commands. (Open the terminal and go to the specified directory. Type in the following commands)

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,185 @@
+$ mkdir textanalyzer
+$ cd textanalyzer
+```
+Now to create the Virtual environment use the code below
````
Describe the virtual environment and why it is needed?

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,185 @@
+Now to create the Virtual environment use the code below
+
+```python
+$ python3.8 -m venv env
````
what are the arguments -m, venv and env doing?

jamessandy replied:
that's how you create a virtual environment in python

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,185 @@
+
+```python
+$ python3.8 -m venv env
+$ source env/bin/activate
````
explain this code as well

jamessandy replied:
this code is activating the virtual env, I don't think this needs explanation to a python-dev

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,185 @@
+st.subheader("Paras Patidar - MLAIT")
+st.write('\n\n')
+```
+The next step will be to write a function that will get all the data from our directory and bring them up,
````
You need to insert an image of your directory structure. It is essential, and we need to ensure we include it whenever required.

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,185 @@
+
+    return data
+```
+If you notice in the function above the first line is *@st.cache*, this mark tells streamlit that whenever the function is called that there are some certain requirements it should check for and they should be met, now create a variable called all_data which should call the *get_all_data* function we just wrote.
````
What are the certain requirements you are referring to? Why is it needed?

jamessandy replied:
the requirements are things that have to do with the compiler and not the developer and we don't need it, it a more like a mark to the compiler.

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,185 @@
+
+```python
+@st.cache
+def preprocessing_data(data):
````
Initially, when you talk about the dataset, using pandas.DataFrame, can you load the dataset into a data frame and perform df.head() operation? It will give a glance at what the dataset is.

jamessandy replied:
the dataset is just a mere 3 text file in a folder and might not need any peak since we're not building a model the conventional way.

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,185 @@
+    return training_data, evaluation_data
+```
+
+Now we are going to create two functions, one will be *preprocessing_step* that will get the data, preprocess it, and then split it, and the second one will be *training_step* that will receive two parameters and vectorize the training text.
````
You also have to explain the machine learning part of it. What is happening in each function? I see you fit the vectorizer with training data but didn't understand what happened with the bernoulliNB function.

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,185 @@
+```python
+streamit run textanalyzer.py
+```
+#### Conclusion
````
Include the final output jpeg/png image. How will it look? Take a screenshot of your terminal and the web app.

lalith1403 reviewed (changes requested):
@jamessandy An amazing article. I loved reading it. For a beginner, most of the code will be difficult to comprehend. Go through the suggestions given and change the article wherever necessary. A couple of images would help enhance readability as well. Excited to read the updated article.

jamessandy commented:
Done @lalith1403

lalith1403 commented:
Just checking if all the images are screenshots and don't need sources.

jamessandy commented:
yeah, they don't need sources

jamessandy commented:
quick question, why was the PR you approved closed ?

lalith1403 commented:
@jamessandy @hectorkambow which one are you referring to? Mention the pr number if possible.

jamessandy commented:
#556

lalith1403 commented:
I think there were a few conflicts. A duplicate branch was created to resolve those. From next time ensure the forked repo is up to date with the eng-ed repo. The article has been published.
 https://www.section.io/engineering-education/building-a-python-package-and-publishing-on-pypi/

lalith1403 reviewed (changes requested):
I found a few grammatical errors, and complex sentences. I suggest you check on hemmingwayapp.com before submitting. Break down complex sentences into smaller sentences. Also, I noticed code snippets beginning with capital letters. I am yet to perform a tech review, but noticed some errors. Once you update the article, I shall proceed with the tech review.(checking of code snippets)

With reference to the new payout structure, the second round of revision results in a drop in payment. Therefore, I suggest you submit a near-to-ready article for reviewing.

This article is still in the first round as the link to the dataset was missing.

jamessandy commented:
Hey @lalith1403 so I've fixed the grammatical errors and some of the complex sentence suggestions given by Hemmingway are out of place given that this is a technical article but I've been able to implement those that within scope. I've also fixed the code snippets too.

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
```diff
@@ -0,0 +1,212 @@
+ 
+#### Building Sentiment Analyzer
+ 
+Now for this beginner’s tutorial, we will be building a sentiment analyzer model and prototype it with streamlit. The first step will be getting a Dataset, and the dataset can be found [HERE](https://drive.google.com/drive/folders/1wk0hkrRhxe6t5g390g5V5O-CUcRB6BZC?usp=sharing), next let’s create a directory and create a virtual environment, open you terminal or command prompt and type in the code ...
```
@jamessandy The intention behind asking you to break down complex sentences into simpler ones is as follows: Consider the following sentence. Instead of the commas, you could replace them with periods. 

In this beginner’s tutorial, we will build a sentiment analyzer model and prototype it with streamlit. The first step will be getting a Dataset, which is available [HERE](https://drive.google.com/drive/folders/1wk0hkrRhxe6t5g390g5V5O-CUcRB6BZC?usp=sharing). Further, let’s create a directory and a virtual environment. Open your terminal or command prompt and type in the code below.

Sentences separated by periods rather than commas. If you can change this then this article will be ready for the final round of review. Also instead of using we _will be building_, use **we build**. Its more concise.

lalith1403 replied:
@jamessandy The intention behind asking you to break down complex sentences into simpler ones is to ensure better readability. Consider the following sentence. Instead of the commas, you could replace them with periods.

In this beginner’s tutorial, we will build a sentiment analyzer model and prototype it with streamlit. The first step will be getting a Dataset, which is available HERE. Further, let’s create a directory and a virtual environment. Open your terminal or command prompt and type in the code below.

Periods rather than commas separate sentences. If you can change this, then this article will be ready for the final round of review. Also, instead of using, _we will be building_, use **we build**. It's more concise.

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,212 @@
+ return data
+```
+ 
+If you notice in the function above the first line is *@st.cache*, this mark tells streamlit that whenever the function is called that there are some certain requirements it should check for and they should be fulfilled, now create a variable called all_data which should call the *get_all_data* function we wrote.
````
Notice the code snippet @st.cache in the first line. This tells streamlit that whenever the function is called, some requirements should be fulfilled. Create a new variable called `all_data.` `all_data` calls the function `get_all_data` we defined earlier.

lalith1403 replied:
Notice the code snippet @st.cache in the first line. This tells streamlit that whenever the function is called, some requirements should be fulfilled. Create a new variable called `all_data.` `all_data` calls the function `get_all_data` we defined earlier.

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
```diff
@@ -0,0 +1,212 @@
+ 
+![Simple](https://github.com/jamessandy/engineering-education/blob/new-article/articles/prototyping-machine-learning-models-with-streamlit/hero.jpg)
+ 
+As Machine learning Engineers or Data scientist, you are used to building cool machine learning models but probably not familiar with building mobile or web apps, or the thought of building an application to serve your model is stressful because of the whole process of building either a web or mobile app so you just leave your model(s) on your notebook without using the weights. Streamlit is a pl ...
```
The first sentence needs to be broken down.

lalith1403 replied:
refer to the examples shown below.

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
```diff
@@ -0,0 +1,212 @@
+ 
+#### Introduction to Streamlit
+ 
+Streamlit is an Open source python framework that enables developers to demo or prototype their ML models and a lot more on a web application, you can simply turn your data science scripts into a website with few lines of code. It saves time, energy, and is also simple to learn, use, and fast too. Now we’ll dive straight into how to install and setup streamlit and then build a text summarizer web ...
```
Streamlit is an open-source Python framework that enables developers to prototype their ML models via web apps. 
Continue after the period. Break it down into simpler sentences.

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,212 @@
+$ source env/bin/activate
+```
+ 
+The first line of code creates the virtual environment while the second activate it. Now copy the dataset as it is in the folder and paste it into this textanalyzer directory, and then create a file and save it as sentiment_analyzer.py, this will be where we will be writing our code to build the streamlit app. The next thing will be to import all the packages we will be using for this tutorial.
````
The second sentence needs to be broken down into 2 sentences.

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,212 @@
+all_data = get_all_data()
+```
+ 
+Next, we create a checkbox using the *st.checkbox* function that will display the dataset, and also the checkbox should have the values from the *all_data* variable we created earlier.
````
dataset, -> dataset. The checkbox should have...

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,212 @@
+ 
+```
+ 
+Now let’s create another checkbox that will display *show_preprocessed_data* and will get the data from the *all_data* function and pass it through the preprocessing data function we wrote above.
````
Instead of using multiple _and(s)_, use periods. Break it down into simpler sentences.

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,212 @@
+ 
+```
+ 
+Now we are going to create two functions, one will be *preprocessing_step* that will get the data, preprocess it, and then split it, and the second one will be *training_step* that will receive two parameters and vectorize the training text.
````
Same thing. Don't use so many commas. A sentence can at max have one or two commas depending on the situation.

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,212 @@
+ return BernoulliNB().fit(training_text,training_result)
+```
+ 
+The next step will be passing the training and evaluation data into the *preprocessing_step* function, choosing a vectorizer( In this case we will be using CountVecotizer), and finally, a classifier variable that will call the *training_step* function and pass in the training_data and vectorizer. And the last line uses BernoulliNB which is a Naives Bayes model it predicts the probability of the i ...
````
Break it down. Also... while mentioning variables using \`\` symbol. Don't use * * to highlight variables.

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,212 @@
+classifier = training_step(training_data,vectorizer)
+```
+ 
+After carrying out the above step the next step will be writing two functions, the first will be *analyze_text* and it will carry out the analysis by taking in the classifier, vectorizer, and text value, and the second function will be called *print_result* and will send a result as positive or negative.
````
break it down.

lalith1403 commented on articles/prototyping-machine-learning-models-with-streamlit/index.md:
````diff
@@ -0,0 +1,212 @@
+ st.write("Press the above button..")
+```
+ 
+The codes above use streamlit *st.text_input* to create a text input form that will display “enter the review, write here”, the if *st.button* will create a button that will display “predict sentiment”, after that the variable result calls the *print_result* function and pass in the parameters. If the whole process the st.sucess(result[1) displays the result else it passes an error message asking ...
````
break it down

lalith1403 reviewed (changes requested):
I have pointed out several places where complex sentences need to be broken down into simpler ones. Avoid using many commas in a sentence. Humans' attention span is limited, and using multiple commas reduces the readability of the article significantly.

jamessandy commented:
just curious, what stage of review is this?

lalith1403 commented:
> just curious, what stage of review is this? @lalith1403 

The first one. Any subsequent reviews will be the second one and so on, that is, if any.

lalith1403 commented:
Have you made the changes @jamessandy? Once done, do leave a comment.

jamessandy commented:
@lalith1403 please I am done

lalith1403 commented:
I found a few grammatical errors on grammarly! A thorough check for spelling mistakes is required! For example. Tabel -> Table.

As a general rule of thumb, it's always good to pass the article through grammarly after every round of editing! Helps avoid minor errors.

jamessandy commented:
Ran the Grammarly check.

lalith1403 commented:
@jamessandy In this article, I have made some changes that we expect the author to make. A ready-to-publish article should have all these traits. Ensure you take into consideration all the points mentioned, and all the changes made.

hectorkambow commented:
@jamessandy 
👋  Goodmorning James - thank you so much for the ML article looking forward to reviewing it and getting it published. - just wanted to give you an FYI this PR has been through an [extensive review and the payout](https://github.com/section-io/engineering-education/blob/master/new_contributors/CONTRIBUTING.md#review-approval--payment-processes) will reflect as such 👍
